src/averageAges.js

```diff
@@ -37,7 +43,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter((person) =>
+    person.sex === 'f'
+      && (withChildren
+        ? people.some(child => child.mother === person.name)
+        : true));
+
+  const womenAverageAge = women.reduce((sum, person) =>
+    sum + (person.died - person.born), 0);
+
+  return womenAverageAge / women.length;
```

Comment:
you have the same logic in men function, please create a function helper for calculating average age